### PR TITLE
add phone mask

### PR DESCRIPTION
**Pull request by SetterVet**

### CODE REVIEW

File: app/assets/javascripts/change_phone.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+function change_phone() {
+        document.getElementById('user_phone').value = mask.unmaskedValue;
```
Comment (afurm) [inline app/assets/javascripts/change_phone.js]: spaces

**Comment by VasylYaremkiv** [comment]
Read style guide for js and update PR according to it:
https://github.com/airbnb/javascript

**Comment by SetterVet** [comment]
fixed @afurm @rubycop @VasylYaremkiv

File: app/assets/javascripts/change_phone.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+function change_phone() {
```
Comment (VasylYaremkiv) [inline app/assets/javascripts/change_phone.js]: Function name: https://github.com/airbnb/javascript#naming--camelCase

File: app/assets/javascripts/change_phone.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
...
+  document.getElementById('user_phone').value = mask.unmaskedValue;
```
Comment (VasylYaremkiv) [inline app/assets/javascripts/change_phone.js]: Spacing: on this you use 2 but in next file 4 spaces

File: app/javascript/packs/application.js [hunk 4]
```diff
@@ -10,5 +12,13 @@ document.addEventListener('DOMContentLoaded', () => {
...
-    })
+    });
+    var phoneMask = new IMask(
+        document.getElementById('user_phone'), {
+            mask: '+{38}(000)000-00-00',
+            lazy: false,
+        });
+
+    window.mask = phoneMask;
+
```
Comment (VasylYaremkiv) [inline app/javascript/packs/application.js]: extra empty line

**Comment by SetterVet** [comment]
fixed @VasylYaremkiv @afurm

File: app/javascript/packs/application.js [hunk 5]
```diff
@@ -1,6 +1,8 @@
 import Vue from 'vue/dist/vue.esm';
 import UsersList from '../components/UsersList.vue'
 import axios from 'axios';
+import IMask from 'imask';
+import change_phone from '../../assets/javascripts/change_phone'
```
Comment (VasylYaremkiv) [inline app/javascript/packs/application.js]: Where do you use `change_phone`? and why it is snake case?

**Comment by SetterVet** [comment]
without global variable and getElementByClassName 
@VasylYaremkiv @afurm @rubycop

File: app/views/layouts/application.html.haml [hunk 6]
```diff
@@ -21,3 +21,4 @@
     %footer.footer
       .container
         %span &copy; CyberCraftInc 2018
+%script{:src => "https://unpkg.com/imask"}
```
Comment (rubycop) [inline app/views/layouts/application.html.haml]: Put this into `assets/application.js`

File: package.json [hunk 7]
```diff
@@ -4,6 +4,7 @@
   "dependencies": {
     "@rails/webpacker": "3.5",
     "axios": "0.18.0",
+    "imask": "^4.1.4",
```
Comment (rubycop) [inline package.json]: Lock version

Comment (SetterVet) [reply package.json]: @rubycop fixed

**Comment by SetterVet** [comment]
@rubycop done

File: app/assets/javascripts/mask.js [hunk 8]
```diff
@@ -0,0 +1,16 @@
+window.onload = function () {
+    Array.from(document.getElementsByClassName('phone-input')).forEach(function (element) {
+        new IMask(element, {
+            mask: '+{38}(000)000-00-00',
+            lazy: false,
+        });
+    });
+};
+
+function changePhone() {
```
Comment (VasylYaremkiv) [inline app/assets/javascripts/mask.js]: Use better name for this function: unmaskPhone

Comment (SetterVet) [reply app/assets/javascripts/mask.js]: @VasylYaremkiv change

**Pull request merged**